tests/php/ORM/DataObjectTest.php

```diff
@@ -2671,4 +2672,36 @@ public function testDBObjectEnum()
         $vals = ['25.25', '50.00', '75.00', '100.50'];
         $this->assertSame(array_combine($vals ?? [], $vals ?? []), $obj->dbObject('MyEnumWithDots')->enumValues());
     }
+
+    public function testGetDatabaseBackedField()
+    {
+        $dataObjectClass = new DataObject();
+        $method = new ReflectionMethod($dataObjectClass, 'getDatabaseBackedField');
+        $method->setAccessible(true);
+        $class = new Team([]);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Captain.FavouriteTeam.Title']);
+        $this->assertSame('Captain.FavouriteTeam.Title', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Captain.ShirtNumber']);
+        $this->assertSame('Captain.ShirtNumber', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Captain.FavouriteTeam.ReturnsNull']);
+        $this->assertSame('Captain.FavouriteTeam', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Founder.FavouriteTeam.Captain.ReturnsNull']);
+        $this->assertSame('Founder.FavouriteTeam.Captain', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['HasOneRelationship.FavouriteTeam.MyTitle']);
+        $this->assertSame('HasOneRelationship.FavouriteTeam', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['SubTeams.Comments.Name.Plain']);
+        $this->assertSame('SubTeams.Comments.Name', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['ReturnsNull']);
+        $this->assertNull($databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['DynamicField']);
+        $this->assertNull($databaseBackedField);
```

Comment:
Please convert this to use a [dataprovider](https://docs.phpunit.de/en/9.6/writing-tests-for-phpunit.html#writing-tests-for-phpunit-data-providers) - that has a couple of benefits:
- It's a lot easier to see what the various test scenarios are
- It's easier to add new test scenarios
- _all_ of the scenarios will be tested, even if one fails early on, which means we have a clear view of what works and what doesn't

Reply:
Updated